airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -404,8 +404,12 @@ def await_pod_start(self, pod: k8s.V1Pod):
     def extract_xcom(self, pod: k8s.V1Pod):
         """Retrieves xcom value and kills xcom sidecar container"""
         result = self.pod_manager.extract_xcom(pod)
-        self.log.info("xcom result: \n%s", result)
-        return json.loads(result)
+        if isinstance(result, str) and result.replace('\n', '') == 'False':
```

Comment:
```suggestion
        if isinstance(result, str) and result.rstrip() == 'False':
```

It may also be worthwile to use a more obscure string (say `__airflow_xcom_result_empty__`?) to make the intent easier to identify.

Reply:
+1 to using rstrip

Reply:
@uranusjr @kaxil I have changed replace() to rstrip()